### DISCUSSION TITLE
Adonisjs Debugbar

### DISCUSSION THREAD

**Discussion by ograna** [adonisjs/core#2604] [Ideas]
Hi there,

I love the Debugbar in Laravel (https://github.com/barryvdh/laravel-debugbar) or the Profiler in Symfony.
Since there is no built-in or third party library out there, i was trying to build this by myself.
But I was not able to hook into all the important application "states" and "events" except database querys.
Am I just to bad to do something like this or is it currently not possible? If not possible, do you think there will be a posibility in the future?

**Comment by MANTENN** [comment]
@ograna you would have to enable debug for the connections, under `config/database.ts`, and then all queries trigger the `Event.on('db:query', callback)` listener.

**Comment by imsyedabdullah** [comment]
I ran into the same problem coming from a Laravel background. Built this package for my own projects, give it a try:
https://github.com/imsyedabdullah/adonis-debugbar
